File: 이건희/level2/스킬트리.py
def solution(skill, skill_trees):
    result = 0
    
    for i in skill_trees:
        order = ''
        
        for j in i:
            if j in skill:
                order += j
                
        if order == skill[:len(order)]:
            result += 1
            
    return result

File: 김찬우/프로그래머스 /Lv.2/스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        int index = 0; 
        boolean check = false;
    
        
        for(int i=0; i<skill_trees.length; i++){
            index =0;
            check = false;
            
            for(int j=0; j<skill_trees[i].length(); j++ ){
                
                String al = String.valueOf(skill_trees[i].charAt(j));
                
                if(skill.indexOf(al) > index){
                    check = true;
                    break;
                    
                }else if(skill.indexOf(al) == -1 ){
                    continue;
                }else if(skill.indexOf(al) <= index){
                    index++;
                }
                
            }
            
            if(!check){
                answer++;
            }
            
        }
        
        return answer;
    }
}
